blog/views.py

```diff
@@ -0,0 +1,38 @@
+from django.shortcuts import redirect
+from django.views.generic import ListView, DetailView
+
+from blog.forms import CommentaryForm
+from blog.models import Post
+
+
+# Create your views here.
+class PostListView(ListView):
+    model = Post
+    paginate_by = 5
+
+
+class PostDetailView(DetailView):
+    model = Post
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['form'] = kwargs.get('form') or CommentaryForm()  # Load the form or pass the one with errors
+        return context
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()  # Get the current post
+        form = CommentaryForm(request.POST)
+
+        # If the user is not authenticated, make the form invalid and add an error
+        if not request.user.is_authenticated:
+            form.add_error(None, "You must be logged in to post a comment.")
```

Comment:
It's good practice to use Django's built-in authentication decorators to handle user authentication. Instead of checking if the user is authenticated within the `post` method, you could use the `@login_required` decorator. This would redirect unauthenticated users to the login page.